### PR TITLE
Blog app - controllers

### PR DESCRIPTION
**Pull request by Gh-Reza**
## Changes:

- Added routes, controllers, and views for given URLs(/,/users/, users/:id, users/:id/posts/, users/:id/posts/:id)

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class PostsController < ApplicationController
+  def posts; end
+
+  def post; end
+end
```
Comment (ghost) [inline app/controllers/posts_controller.rb]: - [ ] 📌 Hey there! Great effort on your PostsController, but there's a small adjustment needed to make it more in line with best practices and the project guidelines. It's recommended to use `index` and `show`  as the action names, like this:

```
class PostsController < ApplicationController
  def index; end
  def show; end
end

```

- [ ] Also remember that you should also adjust your `views` to correspond with the new action names. Make sure that you have both an `index.html.erb` and a `show.html.erb` file in your views directory, so they match the `index` and `show` actions respectively

#### This naming convention is commonly used in Rails and will help your code stay organized and meet the project's specific requirements. Keep up the good work! 😊👍

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class UsersController < ApplicationController
+  def index; end
+
+  def user; end
+end
```
Comment (ghost) [inline app/controllers/users_controller.rb]: - [ ] 📌 Hi there! Your UsersController is looking good, but let's make sure it follows best practices and project guidelines. For this controller, it's recommended to use `index ` and `show`  as the action names, like this:

```
class UsersController < ApplicationController
  def index; end
  def show; end
end

```


- [ ] This naming convention will help keep your code organized and align with the project requirements. Also, don't forget to update your views to match these action names. Keep up the good work! 😊👍

File: config/routes.rb [hunk 3]
```diff
@@ -1,6 +1,12 @@
 Rails.application.routes.draw do
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-
+  root "users#index"
+  get "users/:id", to: "users#user"
+  get "users/:id/posts", to: "posts#posts"
+  get "users/:id/posts/:id", to: "posts#post"
+  resources :users, only: [:index, :show] do
+    resources :posts, only: [:index, :show]
+  end
```
Comment (ghost) [inline config/routes.rb]: #### Optional

- [optional ] 👏 Great job on setting up your routes! 🚀 However, the custom routes you've defined seem to overlap with the standard RESTful routes created by `resources. ` 🔄


#### Your custom routes

```


root "users#index"
  get "users/:id", to: "users#user"
  get "users/:id/posts", to: "posts#posts"
  get "users/:id/posts/:id", to: "posts#post"
```

#### Your standard routes



```
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
```


By removing the custom routes, you'll depend entirely on the RESTful routes offered by resources. This not only simplifies your code but also aligns more closely with RESTful conventions, making your application more predictable and easier to manage. 🛤️🧹

**Review by frank1738 — CHANGES_REQUESTED** [review]
Hi @Gh-Reza , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!




## Highlights

✅ Created  `users_controller` and `posts_controller`
✅ Has users and posts routes added to `config/routes.rb` and maintained a RESTful URL 👍 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions


- [ ] Check the comment below

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AbelG101 — APPROVED** [review]
# Hi @Gh-Reza 👋, 

**Your project is complete!  There is nothing else to say other than... it's time to merge it. 
Congratulations! 🎉 🥳🥇**

## Highlights
- Action method names are properly named ✔
- Corresponding Views are renamed ✔
- Correct use of Git Flow ✔
- All linter checks are passing ✔
- Descriptive PR title with a good summary ✔

## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**